lib/http/HttpServer.ts

```diff
@@ -0,0 +1,108 @@
+import http from 'http';
+import Service from '../service/Service';
+import HttpService from './HttpService';
+import Logger from '../Logger';
+import errors from '../service/errors';
+
+const reqToJson = (req: http.IncomingMessage) => {
+  return new Promise<object>((resolve, reject) => {
+    const body: any[] = [];
+    req.on('data', (chunk) => {
+      body.push(chunk);
+    }).on('end', () => {
+      const bodyStr = Buffer.concat(body).toString();
+      const reqContentLength = parseInt(req.headers['content-length'] || '', 10);
+
+      if (bodyStr.length !== reqContentLength) {
+        reject('invalid content-length header');
+        return;
+      }
+
+      try {
+        resolve(JSON.parse(bodyStr));
+      } catch (err) {
+        reject('cannot parse request json');
+      }
+    });
+  });
+};
+
+class HttpServer {
+  private server: http.Server;
+  private httpService: HttpService;
+
+  constructor(private logger: Logger, service: Service) {
+    this.server = http.createServer(this.requestListener);
+    this.httpService = new HttpService(service);
+  }
+
+  private requestListener: http.RequestListener = async (req, res) => {
+    if (!req.headers['content-length']) {
+      res.writeHead(411);
+      return;
+    }
+
+    let reqJson: any;
+    try {
+      reqJson = await reqToJson(req);
+    } catch (err) {
+      res.writeHead(400);
+      res.end(err);
+      return;
+    }
+
+    let resJson: any;
+    try {
+      switch (req.url) {
+        case '/resolveraiden':
+          resJson = await this.httpService.resolveHashRaiden(reqJson);
+          break;
+        default:
+          res.writeHead(404);
+          res.end();
+      }
+      res.writeHead(200, { 'Content-Type': 'application/json' });
+      res.end(JSON.stringify(resJson));
+    } catch (err) {
+      if (err.code === errors.INVALID_ARGUMENT) {
+        res.writeHead(400);
+        res.end(err.message);
+      } else {
+        res.writeHead(500);
+        res.end(JSON.stringify(err));
+      }
+    }
+  }
+
+  /**
+   * Starts the server and begins listening on the provided port.
+   * @returns true if the server started listening successfully, false otherwise
+   */
+  public listen = async (port: number) => {
+    return new Promise<void>((resolve, reject) => {
+      const listenErrHandler = (err: Error) => {
+        reject(err);
+      };
+
+      this.server.listen(port, '127.0.0.1').once('listening', () => {
```

Comment:
Once this is merged we should also let Offer know that it's not possible to specify the listening interface. I'm assuming this is because there is no authentication.

Reply:
Yes, due to no authentication. Longer term we may want to add some authentication here as well.